openedx_lti_tool_plugin/tests/views/test_launch_view.py

```diff
@@ -0,0 +1,802 @@
+"""Tests for openedx_lti_tool_plugin.views.LtiToolLaunchView."""
+from unittest.mock import MagicMock, patch
+
+from ddt import data, ddt
+from django.test import TestCase
+from django.urls import reverse
+from opaque_keys.edx.keys import CourseKey
+from pylti1p3.exception import LtiException
+
+from openedx_lti_tool_plugin.apps import OpenEdxLtiToolPluginConfig as App
+from openedx_lti_tool_plugin.exceptions import LtiToolLaunchException
+from openedx_lti_tool_plugin.models import LtiProfile
+from openedx_lti_tool_plugin.tests import AUD, COURSE_ID, ISS, SUB, USAGE_KEY
+from openedx_lti_tool_plugin.tests.views.test_views import LtiViewMixin
+from openedx_lti_tool_plugin.views import AGS_CLAIM_ENDPOINT, AGS_SCORE_SCOPE, LtiToolLaunchView
+
+COURSE_KEY = 'random-course-key'
+BASE_LAUNCH_DATA = {'iss': ISS, 'aud': [AUD], 'sub': SUB}
+LTI_PROFILE = 'random-lti-profile'
+PII = {'x': 'x'}
+
+
+class TestLtiToolLaunchViewBase(LtiViewMixin, TestCase):
+    """Test LTI 1.3 platform tool launch view base testcase."""
+
+    def setUp(self):
+        """Test base launch view setup."""
+        super().setUp()
+        self.view_class = LtiToolLaunchView
+
+
+@patch('openedx_lti_tool_plugin.views.DjangoMessageLaunch')
+@patch.object(LtiToolLaunchView, 'get_launch_data')
+@patch.object(LtiToolLaunchView, 'get_identity_claims')
+@patch.object(LtiToolLaunchView, 'check_course_access_permission')
+@patch.object(LtiToolLaunchView, 'get_lti_profile')
+@patch.object(LtiToolLaunchView, 'authenticate_and_login')
+@patch.object(LtiToolLaunchView, 'enroll')
+@patch.object(LtiToolLaunchView, 'handle_ags')
+@patch.object(LtiToolLaunchView, 'get_resource_launch', return_value=(MagicMock(), COURSE_KEY))
+@patch('openedx_lti_tool_plugin.views.DjangoDbToolConf')
+@patch('openedx_lti_tool_plugin.views.DjangoCacheDataStorage')
+@patch.object(CourseKey, 'from_string', return_value=COURSE_KEY)
+class TestLtiToolLaunchViewPost(TestLtiToolLaunchViewBase):
```

Comment:
@alexjmpb There is a test case missing to verify LoggedHttpResponseBadRequest is returned when LtiToolLaunchException is raised